src/ansys/sound/core/sound_composer/track.py

```diff
@@ -0,0 +1,337 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Sound Composer's track."""
+from typing import Union
+import warnings
+
+from ansys.dpf.core import Field
+from matplotlib import pyplot as plt
+import numpy as np
+
+from ansys.sound.core.signal_utilities.apply_gain import ApplyGain
+from ansys.sound.core.sound_composer import (
+    SoundComposerParent,
+    SourceAudio,
+    SourceBroadbandNoise,
+    SourceBroadbandNoiseTwoParameters,
+    SourceHarmonics,
+    SourceHarmonicsTwoParameters,
+    SourceParent,
+    SourceSpectrum,
+)
+from ansys.sound.core.spectral_processing import Filter
+
+from .._pyansys_sound import PyAnsysSoundException, PyAnsysSoundWarning
+
+# Define the typing Union of all possible source types, as a global variable (for typing only).
+AnySourceType = Union[tuple(SourceParent.__subclasses__())]
+
+
+class Track(SoundComposerParent):
+    """Sound Composer's track class.
+
+    This class creates a track for the Sound Composer. A track is made of a source (including its
+    source control) and a filter. A tracks allows the synthesis of the source's sound, filtered
+    with its associated filter.
+    """
+
+    def __init__(
+        self,
+        name: str = "",
+        gain: float = 0.0,
+        source: AnySourceType = None,
+        filter: Filter = None,
+    ):
+        """Class instantiation takes the following parameters.
+
+        Parameters
+        ----------
+        name : str, default: ""
+            Name of the track.
+        gain : float, default: 0.0
+            Gain of the track, in dB.
+        source : SourceSpectrum, SourceBroadbandNoise, SourceBroadbandNoiseTwoParameters, \
+            SourceHarmonics, SourceHarmonicsTwoParameters or SourceAudio, default: None
+            Source of the track.
+        filter : Filter, default: None,
+            Filter of the track.
+        """
+        super().__init__()
+        self.name = name
+        self.gain = gain
+        self.source = source
+        self.filter = filter
+
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
+        if self.source is not None:
```

Comment:
@ansaminard we should not have any checks on source classes types, but ask the Source to describe itself

Reply:
done